src/main/java/com/conveyal/r5/analyst/TravelTimeComputer.java

```diff
@@ -81,18 +85,47 @@ public OneOriginResult computeTravelTimes() throws IOException {
         // Simultaneously we will find stations that allow access to the transit network.
         if (request.transitModes.isEmpty()) {
             // This search will use no transit.
-            //
             // When doing a non-transit walk search, we're not trying to match the behavior of egress and transfer
             // searches which use distance as the quantity to minimize (because they are precalculated and stored as distance,
             // and then converted to times by dividing by speed without regard to weights/penalties for things like stairs).
             // This does mean that walk-only results will not match the walking portion of walk+transit results.
-            sr.timeLimitSeconds = request.maxTripDurationMinutes * 60;
-            sr.streetMode = directMode;
-            sr.quantityToMinimize = StreetRouter.State.RoutingVariable.DURATION_SECONDS;
-            sr.route();
+            if (request.directModes.contains(LegMode.BICYCLE_RENT)) {
+                if (!this.network.streetLayer.bikeSharing) {
+                    LOG.warn("Bike sharing trip requested but no bike sharing stations in the " +
+                        "streetlayer");
+                    travelTimeReducer.finish();
+                    return null;
+                }
+                PointToPointQuery bikeQuery = new PointToPointQuery(network);
```

Comment:
This does look like it would work for your use case and potentially for ours if we wanted to enable bike rental within our spatial analysis system. However it breaks a certain hierarchy that we have in this project, where point to point routing is built on top of the routing components and data structures that were developed for the analysis system. We're hesitant to call "up" to functions in the point-to-point routing system from within the one-to-many code it is built upon. The point-to-point code is expected to be a continuous work in progress to respond to end users' expectations, and we are planning to allow it to become rather ad-hoc and messy. The one-to-many code on the other hand is intended to remain relatively minimalist and "principled" since we're using it in specialist tools that need to run headless in clusters for example.